### PR TITLE
Modernize project

### LINKED ISSUE #13: Modernize project
This project was made without the use of our standard project template.

https://github.com/carver/ethereum-python-project-template

The following should be cleaned up to closer match our *modern* project structure.

- remove `requirements-dev.txt` in favor of using `extras_require` pattern.
- add `classifiers` to `setup.py`
- update the readme to contain the appropriate badgest, title and general structure.
- use the `long_description_content_type` as described here: https://github.com/pypa/setuptools/issues/1236
- update the `tox.ini` `flake8` environment to use the more modern `lint` name as well as the same approach, installing from `extras_require`.
- change licence to be the string `MIT`
- add `python_requires` (and exclude `python==3.5.2`)
- add `Makefile` and `bumpfile` based version management.

### PR DESCRIPTION
**Pull request by iamonuwa**
Fixes #13

### CODE REVIEW

**Comment by iamonuwa** [comment]
@pipermerriam please review

**Comment by pipermerriam** [comment]
Looks like the readme is being taken care of in https://github.com/ethereum/py_ecc/pull/16

**Comment by iamonuwa** [comment]
Just saw that. Will take care of the setup.py then

File: setup.py [hunk 1]
```diff
@@ -14,11 +30,24 @@
...
+    long_description_content_type='Elliptic curve crypto in python including secp256k1 and alt_bn128',
     author='Vitalik Buterin',
     author_email='',
     url='https://github.com/ethereum/py_ecc',
-    license=license,
+    license="MIT",
     packages=find_packages(exclude=('tests', 'docs')),
     install_requires=[
     ],
+    python_requires='>=3.5, <4',
```
[the remark this answers is not in the document]
Comment (iamonuwa) [reply setup.py]: Should also add 2.7 to classifiers?

**Comment by iamonuwa** [comment]
@pipermerriam fixed the changes

**Comment by pipermerriam** [comment]
Rebased onto master

**Pull request merged**